### ISSUE TITLE
impyla does not support Python 3.7

### ISSUE THREAD

**Issue by pauleikis** [cloudera/impyla#312]
Python 3.7 introduced a change which made `async` a reserved keyword

```
>>> from impala.dbapi import connect
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "../.venv/lib/python3.7/site-packages/impala/dbapi.py", line 28, in <module>
    import impala.hiveserver2 as hs2
  File "../.venv/lib/python3.7/site-packages/impala/hiveserver2.py", line 340
    async=True)
        ^
SyntaxError: invalid syntax
```

```
Python 3.7.0
impyla==0.14.1
```

**Comment by ddonetskov** [comment]
Just to confirm that I've faced the same issue with Python 3.7.1 and impyla 0.14.1.

**Comment by xmnlab** [comment]
PR #322 should fix this problem.

**Comment by wesm** [comment]
Is this released yet? I hit this today

**Comment by bharaniabhishek123** [comment]
any workaround for this, I also got this 'invalid syntax' today.

**Comment by dknupp** [comment]
We're probably going to soon submit an "rc" build (pre-release version number, so you'll need to specify it explicitly when pip installing). In order to work with 3.x, we also needed https://github.com/cloudera/impyla/pull/342.

Given all the other changes, this will most likely be 0.15rc1. (Using 0.15a1 has been discussed as well).

**Comment by dknupp** [comment]
Alpha release 0.15a1 is now up on pypi for anyone to try out.
`pip install impyla==0.15a1`

**Comment by tpsimoes** [comment]
> Alpha release 0.15a1 is now up on pypi for anyone to try out.
> `pip install impyla==0.15a1`

dknupp ... thx for your release!

Feedback:

With the following lines:

from impala.dbapi import connect
conn = connect(host='impalasrv-prod', port=21050, kerberos_service_name='impala', auth_mechanism='GSSAPI')

cur = conn.cursor()
cur.execute('select * from db.table limit 2')
cur.fetchall()

It works!

But if we try with :
%load_ext sql
%config SqlMagic.connect_args="{'kerberos_service_name':'impala', 'auth_mechanism':'GSSAPI'}"
%sql impala://user:password@impalasrv-prod:21050/test2

%%sql 
select * from table

it doesnt work!

Thanks

**Comment by dknoxon** [comment]
@dknupp ... This works for us.  Thanks for the fix!

**Comment by dknupp** [comment]
Thanks for the confirmation @tpsimoes and @dknoxon. I'll probably push the actual 0.15.0 release to pypi by the end of the week.

**Comment by tpsimoes** [comment]
> Thanks for the confirmation @tpsimoes and @dknoxon. I'll probably push the actual 0.15.0 release to pypi by the end of the week.

Sure!

@dknupp 

Btw ... it's possible to make impyla compatable with HiveQl Kernel?
Issue with "hiveqlkernel 1.0.13 has requirement thrift-sasl==0.3.*" because impyla requires :"pip install thrift_sasl==0.2.1".

**Comment by arisbw** [comment]
Hi guys, I still got some issues when using it from Python 3.7. I already upgrade impyla to 0.15.0 with error exactly like https://github.com/cloudera/impyla/issues/289
`library that I use:
thrift==0.11.0
thrift-sasl==0.3.0
thriftpy2==0.4.2
`

**Comment by tpsimoes** [comment]
> Hi guys, I still got some issues when using it from Python 3.7. I already upgrade impyla to 0.15.0 but I got an error exactly like #289
> 
> library that I use:
> thrift==0.11.0
> thrift-sasl==0.3.0
> thriftpy2==0.4.2

Hi @arisbw i'v created a tutorial to an Architecture for Cloudera (Cluster) + Jupyter (JupyterHub or JupyterLab) + Kerberos & LDAP with working Kernels (HiveQL, Impyla, PySpark, Python ... just download from my Linkdln or SlideShare ... hope it helps! 

https://www.linkedin.com/in/tpsimoes/

**Comment by AndersonJo** [comment]
I solved this problem in Python 3.7 with the following configuration. 

```
thrift==0.11.0
thrift-sasl==0.2.1
bit-array==0.1.0
impyla==0.15.0
thriftpy==0.3.9
```
